src/docker.go

```diff
@@ -52,48 +52,63 @@ func main() {
 
 	var fetcher raw.Fetcher
 	var docker raw.DockerClient
+
 	if args.Fargate {
-		var err error
-		var metadataBaseURL *url.URL
-		if metadataBaseURL, err = aws.MetadataV4BaseURL(); err != nil {
-			log.Debug("The Metadata endpoint V4 is not available, falling back to V3: %s", err.Error())
-			//If we do not find V4 we fall back to V3
-			metadataBaseURL, err = aws.MetadataV3BaseURL()
-		}
-		exitOnErr(err)
-
-		fetcher, err = aws.NewFargateFetcher(metadataBaseURL)
-		exitOnErr(err)
-
-		docker, err = aws.NewFargateInspector(metadataBaseURL)
-		exitOnErr(err)
+		fetcher, docker = setupFargateFetcher()
 	} else {
-		var tmpDocker *client.Client
-		tmpDocker, err = client.NewClientWithOpts(client.FromEnv, client.WithVersion(args.DockerClientVersion))
-		exitOnErr(err)
-		defer tmpDocker.Close()
-		docker = raw.NewCachedInfoDockerClient(tmpDocker)
-
-		detectedHostRoot, err := raw.DetectHostRoot(args.HostRoot, raw.CanAccessDir)
-		exitOnErr(err)
-
-		cgroupInfo, err := raw.GetCgroupInfo(context.Background(), docker)
-		exitOnErr(err)
-
-		fetcher, err = raw.NewCgroupsFetcher(
-			detectedHostRoot,
-			cgroupInfo,
-			raw.NewPosixSystemCPUReader(),
-			raw.NewNetDevNetworkStatsGetter(),
-		)
-		exitOnErr(err)
+		fetcher, docker = setupCgroupsFetcher(args)
 	}
+
 	sampler, err := nri.NewSampler(fetcher, docker, args)
 	exitOnErr(err)
 	exitOnErr(sampler.SampleAll(context.Background(), i))
 	exitOnErr(i.Publish())
 }
 
+func setupFargateFetcher() (raw.Fetcher, raw.DockerClient) {
+	var err error
+	var metadataBaseURL *url.URL
+	if metadataBaseURL, err = aws.MetadataV4BaseURL(); err != nil {
+		log.Debug("The Metadata endpoint V4 is not available, falling back to V3: %s", err.Error())
+		// If we do not find V4 we fall back to V3
+		metadataBaseURL, err = aws.MetadataV3BaseURL()
+	}
+	exitOnErr(err)
+
+	fetcher, err := aws.NewFargateFetcher(metadataBaseURL)
+	exitOnErr(err)
+
+	docker, err := aws.NewFargateInspector(metadataBaseURL)
+	exitOnErr(err)
+
+	return fetcher, docker
+}
+
+func setupCgroupsFetcher(args config.ArgumentList) (raw.Fetcher, raw.DockerClient) {
+	var err error
+	var dockerClient *client.Client
+	dockerClient, err = client.NewClientWithOpts(client.FromEnv, client.WithVersion(args.DockerClientVersion))
```

Comment:
why not 
```
dockerClient, err := client.NewClientWithOpts(client.FromEnv, client.WithVersion(args.DockerClientVersion))
```

Reply:
I agree but I only wanted to move the code to a different (without changing the current implementation). Let's keep the change of of the scope of this if you don't mind.

Reply:
ah ok I did not notice it was already like that!